ui/prometheus-plugin/src/plugins/JSONSpecEditor.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useRef, useState } from 'react';
+import { TextField } from '@mui/material';
+import { OptionsEditorProps } from '@perses-dev/plugin-system';
+
+export function JSONSpecEditor<T>(props: OptionsEditorProps<T>) {
```

Comment:
Hahaha, love the `TextField` JSON editor as a placeholder. 😗👌  Well done.

Reply:
:) Further enhancements might be optional props like the ones below:

```tsx
<JSONSpecEditor 
   jsonSchema={someSchema} 
   links={[title: "Docs", url: "..." ]} 
   samples={snippets}
/>

```